(PUP-3842,PUP-3844,PUP-3842) Updates for building the puppet windows installer

melissa opened the pull request:
This PR adds updates to what is pulled in and defined for building the puppet MSI. We're changing the name of the installer, adding in mcollective, and using new automation to deal with these changes. These are all targeted at Puppet 4.

ferventcoder commented:
:+1:

puppetcla commented:
CLA signed by all contributors.

ferventcoder commented:
@joshcooper any objections to merging this? I am good with merging it if not.

melissaanne commented:
@ferventcoder, I think @Iristyle wanted to poke at the builds produced from these updates

ferventcoder commented:
right on. Was waiting for another thumbs up before finding the shiny green button.,

Iristyle commented on ext/build_defaults.yaml:
```diff
@@ -14,9 +14,10 @@ yum_host: 'yum.puppetlabs.com'
 build_msi:
   puppet_for_the_win:
-    ref: '2476cd007b1b52fb1b9f22202b56218629626b5c'
+    ref: '898b097a97f6cd72d59290253867ec8002fd7d38'
```
Let's change this to `59031fc0783c70706d1faa774ce4c9cfd74e2965`

puppetcla commented:
Waiting for CLA signature by @melissaanne

@melissaanne - We require a Contributor License Agreement (CLA) for people who contribute to Puppet, but we have an easy click-through license with instructions, which is available at https://cla.puppetlabs.com/

Note: if your contribution is trivial and you think it may be exempt from the CLA, please post a short reply to this comment with details. http://docs.puppetlabs.com/community/trivial_patch_exemption.html

melissa commented:
Dang! The CLA issue is because I just changed by github username from melissaanne to melissa, and changed the CLA signature from my old username to the new one... I'm not sure why it's still looking under my old username :(

ferventcoder commented:
https://github.com/puppetlabs/puppet/commit/b6a477eb1e5c15da17293fde0d23c978ab74125c is the merge commit
